File: src/main/java/student/Student.java
package student;

public class Student {
    String id;
    String name;
    int english;
    int math;

    public Student(String name, int english, int math) {
        this.name = name;
        this.english = english;
        this.math = math;
    }

    public void print(){
        int average = getAverage();
        char grading;
        System.out.print(name + "\t" + english + "\t" + math + "\t" +
                getAverage() + "\t" + (getAverage() >= 60 ? "PASS" : "FAILED"));

        switch (average/10) {
            case 10:
            case 9:
                grading = 'A';
                break;
            case 8:
                grading = 'B';
                break;
            case 7:
                grading = 'C';
                break;
            case 6:
                grading = 'D';
                break;
            default:
                grading = 'F';
        }

        System.out.println("\t" + grading);


/*        char grading = 'F';
        if (average >= 90 && average <= 100) {
            grading = 'A';
        } else if (average >= 80 && average <= 89) {
            grading = 'B';
        } else if (average >= 70 && average <= 79) {
            grading = 'C';
        } else if (average >= 60 && average <= 69) {
            grading = 'D';
        }
        System.out.println("\t" + grading);*/

        /*if (getAverage() >= 60){
            System.out.println("\tPASS");
        } else {
            System.out.println("\tFAILED");
        }*/
    }

    public int highest () {
        // int max = (english > math) ? english : math;
        int max = 0;
        if (english > math){
            max = english;
        } else {
            max = math;
        }
        return max;
    }

    public int getAverage(){
        return (english + math)/2;
    }
}

File: src/main/kotlin/student/Student.kt
package student

import java.util.*

fun main(args: Array<String>) {
//    userInput()
    val stu = Student("Courtney", 80, 77)
    stu.print()
    println("Highest Score: ${stu.highest()}" )
}

class Student(var name: String?, var english: Int, var math: Int){

    fun print(){
        println("$name\t$english\t$math\t${getAverage()}\t${passOrFailed()}\t${grading()}")
    }

    //fun passOrFailed () = if (getAverage() >=60) "PASS" else "FAILED"

    fun passOrFailed() = if (getAverage() >=60) "PASS" else "FAILED"

    fun grading() = when (getAverage()) {
            in 90..100 -> 'A'
            in 80..89 -> 'B'
            in 70..79 -> 'C'
            in 60..69 -> 'C'
            else -> 'F'
        }

    fun getAverage() = (english + math)/2

    fun nameCheck() = println(name?.length)

    fun highest() = if(english > math) {
        println("english")
        english
    } else {
        println("math")
        math
    }
}

private fun userInput() {

    val scanner = Scanner(System.`in`)
    print("Please enter student's name: ")
    var name = scanner.next()
    // var name = null
    print("Please enter student's english: ")
    var english = scanner.nextInt()
    print("Please enter student's math: ")
    var math = scanner.nextInt()
    val stu = Student(name, english, math)
    stu.print()
    stu.nameCheck()

}

